pkg/controller/machinedeployment/controller.go

```diff
@@ -108,9 +108,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler, mapFn handler.ToRequestsFu
 func (r *ReconcileMachineDeployment) getMachineSetsForDeployment(d *v1beta1.MachineDeployment) ([]*v1beta1.MachineSet, error) {
 	// List all MachineSets to find those we own but that no longer match our selector.
 	machineSets := &v1beta1.MachineSetList{}
-	listOptions := &client.ListOptions{Namespace: d.Namespace}
-
-	if err := r.Client.List(context.Background(), listOptions, machineSets); err != nil {
+	if err := r.Client.List(context.Background(), machineSets, func(opt *client.ListOptions) { opt.InNamespace(d.Namespace) }); err != nil {
```

Comment:
This looks weird. Aren't the options here a variadic parameter, and they give you a bunch of functions that return `ListOptionFunc`, so you can do something like what's below?

`r.Client.List(context.Background(), machineSets, client.InNamespace(ns), client.MatchingLabels(labels))`

Reply:
Good point. PR updated.